File: src/1846.py
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: list[int]) -> int:
        arr.sort()
        prev = 0
        for i in range(len(arr)):
            if arr[i] > prev:
                arr[i] = prev + 1
            prev = arr[i]
        return arr[-1]

File: src/1846.cpp
#include <vector>
#include <algorithm>

using namespace std;

class Solution {
public:
    int maximumElementAfterDecrementingAndRearranging(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int prev = 0;
        for (int i = 0; i < arr.size(); ++i) {
            if (arr[i] > prev) {
                arr[i] = prev + 1;
            }
            prev = arr[i];
        }
        return arr.back();
    }
};
